airflow/jobs.py

```diff
@@ -1122,9 +1124,26 @@ def _execute(self):
         )
         self.process = subprocess.Popen(['bash', '-c', command])
         return_code = None
+        last_heartbeat = datetime.now()
+        print("running faster version")
         while return_code is None:
-            self.heartbeat()
+            time_since_last_heartbeat = (datetime.now() -
+                                         last_heartbeat).total_seconds()
+            time_until_next_heartbeat = \
+                int(max(0, self.heartrate - time_since_last_heartbeat))
+
+            # Wait for task to finish running, up until we need to heartbeat.
+            try:
```

Comment:
Is return_code still needed after this change? Could there just be a return after self.process.wait() (and the while converted to while True)?